cards.js

```diff
@@ -17,7 +17,28 @@ function shuffle(array){
 }
 
 function buildDeck(){
+  for(var i = 0; i < values.length; i++){ [i]
+    for(y in suits){ suits[y]
```

Comment:
This might give unexpected results. Typically `for in` loops are used for looping through attributes of an object, and `for( var i = 0; i < len; i++ )` loops are used for looping through items of a known length.

If you're interested in the why, I found this stackoverflow answer that explains it well - http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea